qutebrowser/completion/models/histcategory.py

```diff
@@ -86,11 +72,31 @@ def set_pattern(self, pattern):
         # escape to treat a user input % or _ as a literal, not a wildcard
         pattern = pattern.replace('%', '\\%')
         pattern = pattern.replace('_', '\\_')
-        # treat spaces as wildcards to match any of the typed words
-        pattern = re.sub(r' +', '%', pattern)
-        pattern = '%{}%'.format(pattern)
+        words = ['%{}%'.format(w) for w in pattern.split(' ')]
+
+        wheres = ' AND '.join([
```

Comment:
With `OR`, `:open pull qute` would match `http://www.qutebrowser.org/` because `qute` matches, even if `pull` doesn't.
Good point about the generator though.

Reply:
Ah, makes sense - maybe add a comment with how a generated query would look like somewhere?